main/lsp/requests/hover.cc

```diff
@@ -14,7 +14,7 @@ string methodSignatureString(const core::GlobalState &gs, const core::TypePtr &r
     ;
     while (start != nullptr) {
         auto &dispatchComponent = start->main;
-        if (dispatchComponent.method.exists()) {
+        if (dispatchComponent.method.exists() && !dispatchComponent.method.isSynthetic()) {
```

Comment:
Discussed IRL; I think we shouldn't be sending query responses that refer to synthetic methods, so this should be fixed earlier in the pipeline.

Reply:
Moved this check to `Environment::processBinding`